app/uk/gov/hmrc/tai/service/yourTaxFreeAmount/DescribedYourTaxFreeAmountService.scala

```diff
@@ -56,7 +55,7 @@ class DescribedYourTaxFreeAmountService @Inject()(
       totalTax                <- totalTaxFuture
     } yield {
       totalTax match {
-        case TaiSuccessResponseWithPayload(totalTax: TotalTax) =>
+        case totalTax: TotalTax =>
```

Comment:
I think the match is unnecessary
cc @sergi-hmrc

Reply:
yup this is no longer needed

Reply:
I would suggest using mapN as we did elsewhere: 
```
def taxExplanationPage(): Action[AnyContent] = (authenticate andThen validatePerson).async { implicit request =>
    val nino = request.taiUser.nino
    implicit val user: AuthedUser = request.taiUser
    (
      taxAccountService.totalTax(nino, TaxYear()),
      taxAccountService.taxCodeIncomes(nino, TaxYear()),
      taxAccountService.taxAccountSummary(nino, TaxYear()),
      codingComponentService.taxFreeAmountComponents(nino, TaxYear()),
      taxAccountService.nonTaxCodeIncomes(nino, TaxYear())
      ).mapN {
      case (
        totalTax, Right(taxCodeIncomes),
        TaiSuccessResponseWithPayload(taxAccountSummary: TaxAccountSummary),
        codingComponents,
        TaiSuccessResponseWithPayload(nonTaxCodeIncome: NonTaxCodeIncome)) =>
        val model = DetailedIncomeTaxEstimateViewModel(
          totalTax,
          taxCodeIncomes,
          taxAccountSummary,
          codingComponents,
          nonTaxCodeIncome)
        Ok(detailedIncomeTaxEstimate(model))
      case _ =>
        errorPagesHandler.internalServerError("Failed to fetch total tax details")
    }.recover {
      case NonFatal(e) => errorPagesHandler.internalServerError("Failed to fetch total tax details", Some(e))
    }
  }
  ```

Reply:
the `case _` isn't really necessary as if it fails to pattern match it will go to the recover